packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/DSL_DataSpace_PureGraphManagerExtension.ts

```diff
@@ -34,10 +39,23 @@ export abstract class DSL_DataSpace_PureGraphManagerExtension extends AbstractPu
     actionState?: ActionState,
   ): Promise<DataSpaceAnalysisResult>;
 
-  abstract retrieveDataSpaceAnalysisFromCache(
-    cacheRetriever: () => Promise<PlainObject<DataSpaceAnalysisResult>>,
+  abstract analyzeDataSpaceCoverage(
+    dataSpacePath: string,
+    entitiesRetriever: () => Promise<Entity[]>,
+    cacheRetriever?: () => Promise<PlainObject<DataSpaceAnalysisResult>>,
     actionState?: ActionState,
-  ): Promise<DataSpaceAnalysisResult | undefined>;
+  ): Promise<V1_DataSpaceAnalysisResult>;
+
+  abstract buildDataSpaceAnalytics(
```

Comment:
it's fine to expose this method, by why
```ts
analytics:
      | PlainObject<V1_DataSpaceAnalysisResult>
      | V1_DataSpaceAnalysisResult,
```

Reply:
also, do you want to do this, so we don't have to build the whole `dataSpaceAnalytics`? Else, why can't we just use `analyzeDataSpaceCoverage()`?

Reply:
I think I can just use analyzeDataSpaceCoverage without exposing this